packages/desktop/electron/lib/ledger.js

```diff
@@ -80,20 +80,23 @@ async function signTransactionData(data, bip32Path) {
     const unsignedTransaction = transactionData.getMessageToSign(false)
     const serializedUnsignedTransaction = Buffer.from(RLP.encode(bufArrToArr(unsignedTransaction)))
 
-    const signature = await appEth.signTransaction(bip32Path, serializedUnsignedTransaction, null)
-
-    const signedTransactionObject = Transaction.fromTxData(
-        {
-            ...data,
-            v: '0x' + signature.v,
-            r: '0x' + signature.r,
-            s: '0x' + signature.s,
-        },
-        TX_OPTIONS
-    )
-
-    const serializedSignedTransaction = Buffer.from(RLP.encode(bufArrToArr(signedTransactionObject.raw())))
-    const serializedSignedTransactionString = '0x' + serializedSignedTransaction.toString('hex')
-
-    return { signedTransaction: serializedSignedTransactionString }
+    try {
+        const signature = await appEth.signTransaction(bip32Path, serializedUnsignedTransaction, null)
+        const signedTransactionObject = Transaction.fromTxData(
+            {
+                ...data,
+                v: '0x' + signature.v,
+                r: '0x' + signature.r,
+                s: '0x' + signature.s,
+            },
+            TX_OPTIONS
+        )
+
+        const serializedSignedTransaction = Buffer.from(RLP.encode(bufArrToArr(signedTransactionObject.raw())))
+        const serializedSignedTransactionString = '0x' + serializedSignedTransaction.toString('hex')
+
+        return { signedTransaction: serializedSignedTransactionString }
+    } catch (error) {
+        return { signedTransaction: undefined }
+    }
```

Comment:
Try/catching here defeats the purpose of the try/catch statement on l.28 above.

Reply:
the use for try catch is that if you `Reject` on the ledger device, we want to return undefined to the renderer process, such that we can handle rejecting there

Reply:
Aaah, okay, I see now, where you use this. Maybe we can improve this by sending back the error, so we can make a distinction between the user rejecting or the wrong app being opened on the device. In any case, it's outside the scope of this PR.